### PR TITLE
Expand DCO documentation

### PR DESCRIPTION
**Pull request by hildeweerts**
Instructions for DCO that are less fancy than the one quoted in the current version, but easier to follow for novices (myself included).

### CODE REVIEW

File: docs/contributor_guide/development_process.rst [hunk 1]
```diff
@@ -33,17 +33,57 @@ changing all the docstrings in one PR.
...
+    git config --global alias.c "commit --signoff"
+
+
+Which allows you to commit including a signoff as :code:`git c -m "Your Message"`. 
+
+These instructions were adapted from `this blog post <https://kauri.io/dco-signoff-commiting-code-to-hyperledger-besu/f58190e5e3bc4b1a9ed902bfccfe58b9/a>`_.
 
-Signing off means you need to have your name and email address attached as a
-commit comment, which you can automate using git hooks as shown
-`here <https://stackoverflow.com/questions/15015894/git-add-signed-off-by-line-using-format-signoff-not-working/46536244#46536244>`_.
```
Comment (adrinjalali) [inline docs/contributor_guide/development_process.rst:46]: this actually tells you how to install a git hook which would do it for you. So now when I contribute to `fairlearn`, I actually don't even do the `--signoff` part since it's done as a commit hook.

So I'd suggest leaving this paragraph in as it's the only part which really makes it automated. But I'm happy with the rest.

Comment (hildeweerts) [reply docs/contributor_guide/development_process.rst:46]: True! I added the link. As someone who's not familiar with git hooks, I found the description a bit confusing though. If we could explain it in a bit more detail in the guide, I think that would be a great addition as well!

**Comment by hildeweerts** [comment]
> I'd also wrap the lines at 79 like the old ones if that's okay with you.

Are you referring to the git hooks or something else?

File: docs/contributor_guide/development_process.rst [hunk 2]
```diff
@@ -33,17 +33,58 @@ changing all the docstrings in one PR.
...
 
-Contributions require you to sign a *developer certificate of origin* (DCO)
-declaring that you have the right to, and actually do, grant us the rights to
-use your contribution. For details, visit https://developercertificate.org/.
+All contributions require you to sign the *developer certificate of origin (DCO)*. This is a developer's certification in which you declare that you have the right to, and actually do, grant us the rights to use your contribution. We use the exact same one created and used by the Linux kernel developers. You can read it at https://developercertificate.org.
 
-When you submit a pull request, a DCO-bot will automatically determine whether
-you need to provide a DCO and decorate the PR appropriately (e.g., label,
-comment).
+You sign the DCO by *signing off* every commit comment with your name and email address: *Signed-off-by: Your Name <your.email@example.com>*
```
Comment (adrinjalali) [inline docs/contributor_guide/development_process.rst:38]: same here, and the rest of the doc.

File: docs/contributor_guide/development_process.rst [hunk 3]
```diff
@@ -33,17 +33,58 @@ changing all the docstrings in one PR.
 Developer certificate of origin
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Contributions require you to sign a *developer certificate of origin* (DCO)
-declaring that you have the right to, and actually do, grant us the rights to
-use your contribution. For details, visit https://developercertificate.org/.
+All contributions require you to sign the *developer certificate of origin (DCO)*. This is a developer's certification in which you declare that you have the right to, and actually do, grant us the rights to use your contribution. We use the exact same one created and used by the Linux kernel developers. You can read it at https://developercertificate.org.
```
[the remark this answers is not in the document]
Comment (hildeweerts) [reply docs/contributor_guide/development_process.rst:36]: Ooh, I see!

For some reason, my brain processed your comment as changing something at the 79th line of the document. I will change it!

File: docs/contributor_guide/development_process.rst [hunk 4]
```diff
@@ -86,7 +142,8 @@ command
 Note that the Fairlearn dashboard is built using nodejs and requires
 additional steps. To build the Fairlearn dashboard after making changes to it,
 `install Yarn <https://yarnpkg.com/lang/en/docs/install>`_, and then run the
-`widget build script <https://github.com/fairlearn/fairlearn/tree/master/scripts/build_widget.py>`_.
+`widget build script <https://github.com/fairlearn/fairlearn/tree/master/
+scripts/build_widget.py>`_.
```
Comment (riedgar-ms) [inline docs/contributor_guide/development_process.rst:146]: Could you undo this change? I think it's OK to overrun 79 characters for things like URLs which are longer than the limit.

File: docs/contributor_guide/development_process.rst [hunk 5]
```diff
@@ -26,24 +26,80 @@ If you're a new contributor please also add yourself to AUTHORS.md.
 Docstrings should follow
 `numpydoc format <https://numpydoc.readthedocs.io/en/latest/format.html>`_.
 This is a
-`recent decision by the community <https://github.com/fairlearn/fairlearn/issues/314>`_.
+`recent decision by the community
+<https://github.com/fairlearn/fairlearn/issues/314>`_.
```
Comment (riedgar-ms) [inline docs/contributor_guide/development_process.rst:30]: Can you undo this change?

**Comment by hildeweerts** [comment]
I was actually surprised that the wrapped links did work (at least in my local build of the docs). I think I got all URLs on single lines now!

**Comment by hildeweerts** [comment]
Nope, I'm not authorized.
